src/api/config/initializers/wrap_parameters.rb

```diff
@@ -2,69 +2,40 @@
 
 # This file contains settings for ActionController::ParamsWrapper which
 # is enabled by default.
-require 'active_support/core_ext/hash/conversions'
 require 'action_dispatch/http/request'
 require 'active_support/core_ext/hash/indifferent_access'
 require 'api_error'
 
 # Disable all default rails parameter parsing
+# Context: https://github.com/rails/rails/blob/39413de44c0e2c0dd2d964be5985b03d8f968a7b/guides/source/3_1_release_notes.md#configinitializerswrap_parametersrb
 ActiveSupport.on_load(:action_controller) do
   wrap_parameters(false) if respond_to?(:wrap_parameters)
 end
 
 # Disable root element in JSON by default.
+# Context: https://github.com/rails/rails/blob/39413de44c0e2c0dd2d964be5985b03d8f968a7b/guides/source/3_1_release_notes.md#configinitializerswrap_parametersrb
 ActiveSupport.on_load(:active_record) do
   self.include_root_in_json = false
 end
 
-OBSApi::Application.config.middleware.delete 'ActionDispatch::ParamsParser'
+# Redefine JSON parser to use ActiveSupport::HashWithIndifferentAccess
+# Define XML parser, since Rails' default parsers only include JSON. This XML parser also uses ActiveSupport::HashWithIndifferentAccess
+# Context: https://github.com/rails/rails/blob/04972d9b9ef60796dc8f0917817b5392d61fcf09/actionpack/lib/action_dispatch/http/parameters.rb#L35-L46
+original_parsers = ActionDispatch::Request.parameter_parsers
 
-# custom params parser (modified form of ActionDispatch::ParamsParser)
-
-class MyParamsParser
-  class ParseError < APIError
-  end
-
-  def initialize(app)
-    @app = app
-  end
-
-  def call(env)
-    params = parse_parameters(env)
-    env['action_dispatch.request.request_parameters'] = params if params
-
-    env['HTTP_ACCEPT'] ||= 'application/xml'
-    @app.call(env)
-  end
-
-  def parse_parameters(env)
-    request = ActionDispatch::Request.new(env)
-
-    return false if request.content_length.zero?
+json_parser = lambda do |raw_post|
+  original_parsers[Mime[:json].symbol].call(raw_post).with_indifferent_access
+end
 
-    case request.content_mime_type
-    when Mime[:json]
-      begin
-        data = ActiveSupport::JSON.decode(request.raw_post)
-      rescue JSON::ParserError => e
-        Rails.logger.info "failed to parse JSON: #{e.message}"
-        return false
-      end
-      request.body.rewind if request.body.respond_to?(:rewind)
```

Comment:
Rewinding the request body isn't needed anymore, this is handled in [raw_post](https://github.com/rails/rails/blob/39413de44c0e2c0dd2d964be5985b03d8f968a7b/actionpack/lib/action_dispatch/http/request.rb#L332-L341), which is what the lambdas defined in this initializer will receive when they are called.